src/ageAssurance/components/NoAccessScreen.tsx

```diff
@@ -0,0 +1,340 @@
+import {AgeAssuranceRedirectDialog} from '#/components/ageAssurance/AgeAssuranceRedirectDialog'
+import {Button, ButtonIcon, ButtonText} from '#/components/Button'
+import {useDialogControl} from '#/components/Dialog'
+import * as Dialog from '#/components/Dialog'
+import {BirthDateSettingsDialog} from '#/components/dialogs/BirthDateSettings'
+import {DeviceLocationRequestDialog} from '#/components/dialogs/DeviceLocationRequestDialog'
+import {Full as Logo} from '#/components/icons/Logo'
+import {ShieldCheck_Stroke2_Corner0_Rounded as ShieldIcon} from '#/components/icons/Shield'
+import {createStaticClick, SimpleInlineLinkText} from '#/components/Link'
+import {Outlet as PortalOutlet} from '#/components/Portal'
+import * as Toast from '#/components/Toast'
+import {Text} from '#/components/Typography'
+import {BottomSheetOutlet} from '#/../modules/bottom-sheet'
+import {useAgeAssurance} from '#/ageAssurance'
+import {useAgeAssuranceDataContext} from '#/ageAssurance/data'
+import {useComputeAgeAssuranceRegionAccess} from '#/ageAssurance/useComputeAgeAssuranceRegionAccess'
+import {
+  isLegacyBirthdateBug,
+  useAgeAssuranceRegionConfig,
+} from '#/ageAssurance/util'
+import {useDeviceGeolocationApi} from '#/geolocation'
+
+export function NoAccessScreen() {
+  const t = useTheme()
+  const {_} = useLingui()
+  const {gtPhone} = useBreakpoints()
+  const insets = useSafeAreaInsets()
+  const birthdayControl = useDialogControl()
+  const {data} = useAgeAssuranceDataContext()
+  const region = useAgeAssuranceRegionConfig()
+  const isBirthdateUpdateAllowed = useIsBirthDateUpdateAllowed()
+  const {logoutCurrentAccount} = useSessionApi()
+
+  const isAARegion = !!region
+  const hasDeclaredAge = data?.declaredAge !== undefined
+  const canUpdateBirthday =
+    isBirthdateUpdateAllowed || isLegacyBirthdateBug(data?.birthdate || '')
+
+  useEffect(() => {
+    // just counting overall hits here
+    logger.metric(`blockedGeoOverlay:shown`, {})
+  }, [])
+
+  const textStyles = [a.text_md, a.leading_normal]
+
+  const blocks = [
+    _(
+      msg`You are accessing Bluesky from a region that legally requires us to verify your age before allowing you to access the app.`,
+    ),
+  ]
+
+  const onPressLogout = useCallback(() => {
+    if (isWeb) {
+      // We're switching accounts, which remounts the entire app.
+      // On mobile, this gets us Home, but on the web we also need reset the URL.
+      // We can't change the URL via a navigate() call because the navigator
+      // itself is about to unmount, and it calls pushState() too late.
+      // So we change the URL ourselves. The navigator will pick it up on remount.
+      history.pushState(null, '', '/')
+    }
+    logoutCurrentAccount('AgeAssuranceNoAccessScreen')
+  }, [logoutCurrentAccount])
+
+  const birthdayUpdateText = canUpdateBirthday ? (
+    <Text style={[textStyles]}>
+      <Trans>
+        If your birth date is not accurate, you can update it by{' '}
```

Comment:
```suggestion
        If your birthdate is not accurate, you can update it by{' '}
```
nit: The strings are currently inconsistent, but I suggest it might be a good idea to standardize on either `birthdate` or `date of birth`.

Reply:
Eh, I think it's fine